themes/10up-block-theme/includes/core.php

```diff
@@ -0,0 +1,190 @@
+	add_action( 'after_setup_theme', 'TenupBlockTheme\Core\theme_setup' );
+	add_action( 'init', 'TenupBlockTheme\Core\scripts' );
+	add_action( 'wp_enqueue_scripts', 'TenupBlockTheme\Core\styles' );
+	add_action( 'enqueue_block_editor_assets', 'TenupBlockTheme\Core\editor_style_overrides' );
+	add_action( 'wp_head', 'TenupBlockTheme\Core\js_detection', 0 );
+	add_action( 'wp_head', 'TenupBlockTheme\Core\scrollbar_detection', 0 );
+	add_action( 'wp_head', 'TenupBlockTheme\Core\embed_ct_css', 0 );
+	add_action( 'init', 'TenupBlockTheme\Core\register_all_icons', 10 );
+}
+
+/**
+ * Makes Theme available for translation.
+ *
+ * Translations can be added to the /languages directory.
+ * If you're building a theme based on "tenup-theme", change the
+ * filename of '/languages/TenupBlockTheme.pot' to the name of your project.
+ *
+ * @return void
+ */
+function i18n() {
+	load_theme_textdomain( 'tenup-theme', TENUP_BLOCK_THEME_PATH . '/languages' );
+}
+
+/**
+ * Sets up theme defaults and registers support for various WordPress features.
+ */
+function theme_setup() {
+	add_theme_support( 'editor-styles' );
+	add_editor_style( '/dist/css/frontend.css' );
+	remove_theme_support( 'core-block-patterns' );
+}
+
+/**
+ * Enqueue scripts for front-end.
+ *
+ * @return void
+ */
+function scripts() {
+
+	wp_enqueue_script(
+		'frontend',
+		TENUP_BLOCK_THEME_TEMPLATE_URL . '/dist/js/frontend.js',
+		get_asset_info( 'frontend', 'dependencies' ),
+		get_asset_info( 'frontend', 'version' ),
+		[
+			'strategy' => 'defer',
+		]
+	);
+}
+
+
+/**
+ * Enqueue styles for front-end.
+ *
+ * @return void
+ */
+function styles() {
+	wp_enqueue_style(
+		'tenup-theme-styles',
+		TENUP_BLOCK_THEME_TEMPLATE_URL . '/dist/css/frontend.css',
+		[],
+		get_asset_info( 'frontend', 'version' )
+	);
+}
+
+/**
+ * Enqueue styles for editor only.
+ *
+ * @return void
+ */
+function editor_style_overrides() {
+	wp_enqueue_style(
+		'tenup-theme-editor-style-overrides',
+		TENUP_BLOCK_THEME_TEMPLATE_URL . '/dist/css/editor-style-overrides.css',
+		[],
+		TENUP_BLOCK_THEME_VERSION
+	);
+
+	wp_enqueue_script(
+		'tenup-theme-block-extensions',
+		TENUP_BLOCK_THEME_TEMPLATE_URL . '/dist/js/block-extensions.js',
+		get_asset_info( 'block-extensions', 'dependencies' ),
+		get_asset_info( 'block-extensions', 'version' ),
+		true
+	);
+}
+
+/**
+ * register all icons located in the dist/svg folder
+ */
+function register_all_icons() {
+	if ( ! function_exists( '\UIKitCore\Helpers\register_icons' ) ) {
+		return;
+	}
+
+	$icon_paths = glob( TENUP_BLOCK_THEME_DIST_PATH . 'svg/*.svg' );
+	$icons      = array_map(
+		function ( $icon_path ) {
+			$icon_name = preg_replace( '#\..*$#', '', basename( $icon_path ) );
+
+			return new \UIKitCore\Icon(
+				$icon_name,
+				ucwords( str_replace( '-', ' ', $icon_name ) ),
+				$icon_path
+			);
+		},
+		$icon_paths
+	);
+
+	\UIKitCore\Helpers\register_icons(
+		[
+			'name'  => 'tenup',
+			'label' => 'Theme Icons',
+			'icons' => $icons,
+		]
+	);
+}
```

Comment:
Very torn if we want this here. 

I think every single project we build needs to have a solution for icons. But this makes it dependent on UI Kit. So not sure what the best option is.

Reply:
@fabiankaegy 

I'd keep it. I feel like this makes sense - if we are using UI Kit - "it should just work" and your early escape resolves if not. 

I would add more detail in comments regarding the dependency on UI Kit. 

If we find helpful, someone can add an issue to utilize a more agnostic approach option within this function.